plugins/SpectrumAnalyzer/SaControls.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * SaControls.h - declaration of SaControls class.
+ *
+ * Copyright (c) 2019 Martin Pavelek <he29/dot/HS/at/gmail/dot/com>
+ *
+ * This file is part of LMMS - https://lmms.io
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program (see COPYING); if not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301 USA.
+ *
+ */
+
+#ifndef SACONTROLS_H
+#define SACONTROLS_H
+
+#include <QPainter>
+
+#include "ComboBoxModel.h"
+#include "EffectControls.h"
+
+//#define SA_DEBUG 1	// define SA_DEBUG to enable performance measurements
+
+// Frequency ranges (in Hz).
+// Full range is defined by LOWEST_LOG_FREQ and current sample rate.
+const int LOWEST_LOG_FREQ = 10;		// arbitrary low limit for log. scale, >1
+
+enum FREQUENCY_RANGES
```

Comment:
I specifically used all-caps to make sure it is distinct from all members and local variables (inspired by the window list in fft_helpers), so I will change it when a better place for these constants is found.

Reply:
So, what about this? Should I change the format or leave it for now, to be resolved when the enums and constants are moved?
I'm not sure how to format the constants, though, since if I just use lower case, it will only become confusing (as variables without prefix are usually local variables).

Reply:
Hmm the [coding conventions](https://github.com/LMMS/lmms/wiki/Coding-conventions) say enums should be treated in upper camel case with leading capital letter:

```
enum MyEnum
{
	ListItem1,
	ListItem2,
	...
} ;
```

The leading capital letter should help you differing those constants from local variables.

If you are sure those variables will be moved anyways, you can delay the casing for now. If you think they might stay here forever, I'd rather change the casing.

Reply:
The enums are clear to me, I meant the range constants. If all-caps is supposed to be reserved only for macros, they should be reformatted to.
And there is not really any convention talking about global constants (probably because they are usually not the best idea to use). The fft_helpers uses all-caps for a global "FFT_BUFFER_SIZE" constant, so I just used the same style (and also because it makes it clear they are not ordinary local constants).
But the fft_helpers used an outdated style anyway, so it is not the best example for sure..

Reply:
Oh, sure, globals being "evil" might be the reason they are not listed in the coding conventions. In this case, I'm also for leaving them all caps.

Reply:
All right. So I'm marking this resolved for now.